### PR TITLE
Support multimodal inputs for inference

### PR DESCRIPTION
**Pull request by isaac-chung**
Support multimodal inputs for inference without using private methods

```diff
from clarifai.client.model import Model
from clarifai.client.input import Inputs

prompt = "What time of day is it?"
image_url = "https://samples.clarifai.com/metro-north.jpg"
inference_params = dict(temperature=0.2, max_tokens=100)

-text_pb = Inputs.get_input_from_bytes("", text_bytes=prompt.encode()).data.text
-image_pb = Inputs.get_input_from_url("", image_url= image_url).data.image

# Model Predict
model_prediction = Model("https://clarifai.com/openai/chat-completion/models/openai-gpt-4-vision").
-    predict(inputs = [Inputs._get_proto(input_id="",imagepb=image_pb,text_pb=text_pb)],
+    predict(inputs = [Inputs.get_multimodal_input(input_id="",image_url=image_url, raw_text=prompt)],
inference_params=inference_params)

print(model_prediction.outputs[0].data.text.raw)
```

### CODE REVIEW

File: clarifai/client/input.py [hunk 1]
```diff
@@ -280,6 +280,36 @@ def get_text_input(input_id: str, raw_text: str, dataset_id: str = None,
...
+    Returns:
+        Input: An Input object for the specified input ID.
+
+    Example:
+        >>> from clarifai.client.input import Inputs
+        >>> input_protos = Inputs.get_multimodal_input(input_id = 'demo', raw_text = 'What time of day is it?', image_url='https://samples.clarifai.com/metro-north.jpg')
+    """
+    image_pb = resources_pb2.Image(base64=image_bytes) if image_bytes and not image_url else None
+    image_pb = resources_pb2.Image(url=image_url) if image_url and not image_bytes else None
+    text_pb = resources_pb2.Text(raw=raw_text)
```
[the remark this answers is not in the document]
Comment (isaac-chung) [reply clarifai/client/input.py:309]: We can, but why instead? I think raw_text will be used more often. Most examples we have, even examples you provided, all use raw text. Is there a reason for adding those that I'm missing other than consistency?

File: clarifai/client/input.py [hunk 2]
```diff
@@ -280,6 +280,40 @@ def get_text_input(input_id: str, raw_text: str, dataset_id: str = None,
...
+        **kwargs: Additional keyword arguments to be passed to the Input
+
+    Returns:
+        Input: An Input object for the specified input ID.
+
+    Example:
+        >>> from clarifai.client.input import Inputs
+        >>> input_protos = Inputs.get_multimodal_input(input_id = 'demo', raw_text = 'What time of day is it?', image_url='https://samples.clarifai.com/metro-north.jpg')
+    """
+    # Will only use URL if both bytes and URL are supplied.
```
Comment (sainivedh) [inline clarifai/client/input.py:309]: Nit: Maybe a Usererror if both url/bytes are given?

File: clarifai/client/input.py [hunk 3]
```diff
@@ -280,6 +280,40 @@ def get_text_input(input_id: str, raw_text: str, dataset_id: str = None,
...
+    image_pb = resources_pb2.Image(base64=image_bytes) if image_bytes else None
+    image_pb = resources_pb2.Image(url=image_url) if image_url else None
+    text_pb = resources_pb2.Text(raw=text_bytes) if text_bytes else None
```
Comment (sainivedh) [inline clarifai/client/input.py:312]: This proto is being overwritten by raw_text, no use of giving text_bytes

**Review by sainivedh — APPROVED** [review]
lgtm

**Pull request merged**